book/glossary.md

```diff
@@ -23,13 +23,26 @@ pages.
 __HTTPS__: A variant of HTTP that uses public-key cryptography for
 network security.
 
+__Hypertext__: A non-linear form of information comprised of multiple documents
+connected with contextual links.
+
+__JavaScript__: The scripting language for the web. (WebAssembly also now
+exists but is much less common.)
+
 __URL__: Uniform Resource Locator. The name used to refer uniquely to a web
 page.
 
 __Rendering engine__: The part of a Web Browser concerned with drawing a web
 page to the screen and interacting with it. There are three rendering engines
 actively maintained today: Chromium, WebKit and Gecko.
 
+__Script__: A piece of code that extends a web page with more functionality,
+usually written in __JavaScript__.
+
+__Web Security__: The property of the web being safe to load a web page in a
+browser without harm (such as malicious software reading private data or 
+corrupting computer state) will result.
```

Comment:
Hmm, this seems like an odd definition. Like—this is a kind of web security possibly exclusive of web application security.

Reply:
I'm not even sure I *can* come up with a good definition. Best I can do is:

> The ability of the web (or individual browsers or applications that are part of it) to be used without causing unintentional harm. There are lots of different aspects of security: browser security (the computer system a browser is running on can't be harmed by it), web application security (a web application can't be harmed by its users), privacy (a third party can't harm a web user by observing their use of the web), and many others.

Reply:
If you don't like this one I say we just don't define this.